File: vol1/java-examples/src/main/java/com/heatonresearch/aifh/error/ErrorCalculationMSE.java
package com.heatonresearch.aifh.error;

/**
 * Created with IntelliJ IDEA.
 * User: jheaton
 * Date: 7/27/13
 * Time: 8:26 AM
 * To change this template use File | Settings | File Templates.
 */
public class ErrorCalculationMSE extends AbstractErrorCalculation {

    /**
     * Calculate the error with MSE.
     *
     * @return The current error for the neural network.
     */
    @Override
    public final double calculate() {
        if (this.setSize == 0) {
            return Double.POSITIVE_INFINITY;
        }
        final double err = this.globalError / this.setSize;
        return err;

    }
}

File: vol3/vol3-csharp-examples/AIFH-Vol3-Core/Core/Error/ErrorCalculationMSE.cs

namespace AIFH_Vol3.Core.Error
{
    /// <summary>
    /// Calculates the error as the average of the sum of the squared differences between the actual and ideal vectors.
    /// This is the most commonly used error calculation technique in this book.
    ///
    /// http://www.heatonresearch.com/wiki/Mean_Square_Error
    /// </summary>
    public class ErrorCalculationMSE : AbstractErrorCalculation
    {
        /// <summary>
        /// Calculate the error with MSE.
        /// </summary>
        /// <returns>The current error.</returns>
        public override double Calculate()
        {
            if (SetSize == 0)
            {
                return double.PositiveInfinity;
            }
            return GlobalError / SetSize;

        }
    }
}
